src/Raven.Server/Smuggler/Documents/MultiShardedDestination.cs

```diff
@@ -162,8 +164,21 @@ protected ShardedActions(ShardedDatabaseContext databaseContext, Dictionary<int,
                 _rtnCtx = DatabaseContext.AllocateContext(out _context);
             }
 
-            public JsonOperationContext GetContextForNewCompareExchangeValue() => _context;
-            public JsonOperationContext GetContextForNewDocument() => _context;
+            public JsonOperationContext GetContextForNewCompareExchangeValue() => GetContextForNewDocument();
+            public JsonOperationContext GetContextForNewDocument()
+            {
+                if (_context.AllocatedMemory > 16 * 1024 * 1024)
```

Comment:
The problem here was that if we use the same context we will end up in this situation:
![high-mem](https://issues.hibernatingrhinos.com/api/files/74-283608?sign=MTY3NDE3MjgwMDAwMHwyMC04MTl8NzQtMjgzNjA4fGxyRmJXTFJJTG03NWpGZVo1WVAtZi02RjVs%0D%0ARm9ybW1EWVZGamNQM1JtYncNCg%0D%0A&updated=1673674640121)

Because we will not dispose the modifier here:
https://github.com/ravendb/ravendb/blob/ca9ae26f52f633118c0bede47e5aeb9b8b5da640/src/Raven.Server/Smuggler/Documents/StreamSource.cs#L1337

This is probably not the best solution, and we would to want to renew/reset the `BlittableMetadataModifier`, but I will let aviv to figure this out :)

Reply:
Why aren't we disposing it?

Reply:
we are not disposing it here because after renewing the context, the old context is used [here](https://github.com/ravendb/ravendb/blob/v5.4/src/Raven.Server/Smuggler/Documents/StreamSource.cs#L1452) in `modifier.Dispose()` which releases the allocations from old context